policy-controller/k8s/index/src/outbound_index.rs

```diff
@@ -0,0 +1,361 @@
+        for service in self.service_routes.values_mut() {
+            service.delete(&name);
+        }
+    }
+
+    fn service_routes_or_default(&mut self, service_port: ServicePort) -> &mut ServiceRoutes {
+        let authority = format!(
+            "{}.{}.svc.{}:{}",
+            service_port.service, self.namespace, self.cluster_domain, service_port.port
+        );
+        self.service_routes
+            .entry(service_port.clone())
+            .or_insert_with(|| {
+                let opaque = match self.services.get(&service_port.service) {
+                    Some(svc) => svc.opaque_ports.contains(&service_port.port),
+                    None => false,
+                };
+                let (sender, _) = watch::channel(OutboundPolicy {
+                    http_routes: Default::default(),
+                    authority,
+                    namespace: self.namespace.to_string(),
+                    opaque,
+                });
+                ServiceRoutes {
+                    routes: Default::default(),
+                    watch: sender,
+                    opaque,
+                }
+            })
+    }
+
+    fn convert_route(&self, route: HttpRoute) -> Result<OutboundHttpRoute> {
+        let hostnames = route
+            .spec
+            .hostnames
+            .into_iter()
+            .flatten()
+            .map(convert::host_match)
+            .collect();
+
+        let rules = route
+            .spec
+            .rules
+            .into_iter()
+            .flatten()
+            .map(|r| self.convert_rule(r))
+            .collect::<Result<_>>()?;
+
+        let creation_timestamp = route.metadata.creation_timestamp.map(|Time(t)| t);
+
+        Ok(OutboundHttpRoute {
+            hostnames,
+            rules,
+            creation_timestamp,
+        })
+    }
+
+    fn convert_rule(&self, rule: HttpRouteRule) -> Result<OutboundHttpRouteRule> {
+        let matches = rule
+            .matches
+            .into_iter()
+            .flatten()
+            .map(InboundRouteBinding::try_match)
+            .collect::<Result<_>>()?;
+
+        let backends = rule
+            .backend_refs
+            .into_iter()
+            .flatten()
+            .filter_map(|b| self.convert_backend(b))
+            .collect();
+        Ok(OutboundHttpRouteRule { matches, backends })
+    }
+
+    fn convert_backend(&self, backend: HttpBackendRef) -> Option<Backend> {
+        backend.backend_ref.map(|backend| {
+            let port = backend.inner.port.unwrap_or_else(|| {
+                tracing::warn!(?backend, "missing port in backend_ref");
+                u16::default()
+            });
+            let dst = WeightedDst {
+                weight: backend.weight.unwrap_or(1).into(),
+                authority: fmt::format(format_args!(
+                    "{}.{}.svc.{}:{}",
+                    backend.inner.name, self.namespace, self.cluster_domain, port
+                )),
```

Comment:
is this meaningfully different from
```suggestion
                authority: format!(
                    "{}.{}.svc.{}:{}",
                    backend.inner.name, self.namespace, self.cluster_domain, port
                ),
```